setup.py

```diff
@@ -16,6 +20,51 @@
 with open("README.md", "r") as f:
     readme = f.read()
 
+
+class DevelopCommand(develop):
+    """Pre-installation for development mode."""
+
+    def run(self):
+        get_mesa_viz_files()
+        develop.run(self)
+
+
+class InstallCommand(install):
+    """Pre-installation for installation mode."""
+
+    def run(self):
+        get_mesa_viz_files()
+        install.run(self)
+
+
+def get_mesa_viz_files():
+    get_viz_server_file(
+        package="mesa", server_file="visualization/ModularVisualization.py"
+    )
+    get_mesa_template(package="mesa", base_dir="visualization/templates", path="")
+
+
+def get_viz_server_file(package, server_file):
+    viz_server_file = pkgutil.get_data(package, server_file)
+    with open(os.path.join("mesa_geo", server_file), "wb") as server_file:
+        server_file.write(viz_server_file)
+
+
+def get_mesa_template(package, base_dir, path):
```

Comment:
Much simpler to do recursive copy with `shutil.copytree`. See https://docs.python.org/3/library/shutil.html#shutil.copytree.

Reply:
oh, I didn't know this function. It's indeed much simpler. The problem is:

```bash
error: [Errno 2] No such file or directory: 'mesa/visualization/templates'
```

I guess it's due to us trying to access files from another package (mesa) that don't exist in mesa_geo. `pkgutil` and `pkg_resources` don't have this issue.

There is a workaround like this to make it work:

```python
def get_mesa_viz_files():
    ...
    get_mesa_template(package="mesa", template_dir="visualization/templates")
    ...


def get_mesa_template(package, template_dir):
    pkg_dir = sys.modules[package].__path__[0]
    shutil.copytree(
        os.path.join(pkg_dir, template_dir),
        os.path.join("mesa_geo", template_dir),
        dirs_exist_ok=True,
    )
```

Btw the `dirs_exist_ok` argument is only available since [Python 3.8](https://docs.python.org/3/whatsnew/3.8.html#shutil).

Would this be fine?

Reply:
You can do `os.makedirs(the_dir, exists_ok=True)` to ensure the dir exists. But I think it's fine for mesa-geo to have the Python requirement to be `>=3.8`, different from Mesa's `>=3.7'. 3.7 is close to EOL I think?

Reply:
Security support for 3.7 ends in a year or so: https://endoflife.date/python

The `dirs_exist_ok` actually did the opposite: without it, `shutil` will raise an error if the destination already exists. Here it is needed, because directories such as `visualization/templates/js` do exist in mesa-geo.